src/codewhisperer/util/supplementalContext/codeParsingUtil.ts

```diff
@@ -68,29 +67,26 @@ export function countSubstringMatches(arr1: string[], arr2: string[]): number {
     return count
 }
 
-export async function isTestFile(editor: vscode.TextEditor, dependencyGraph: DependencyGraph): Promise<boolean> {
-    const languageConfig = utgLanguageConfigs[editor.document.languageId]
-    if (!languageConfig) {
-        // We have enabled the support only for python and Java for this check
-        // as we depend on Regex for this validation.
-        return false
-    }
+export async function isTestFile(
+    filePath: string,
+    languageConfig: { languageId: string; dependencyGraph?: DependencyGraph; fileContent?: string }
+): Promise<boolean> {
+    const pathContainsTest = filePath.includes(`tests/`) || filePath.includes('test/') || filePath.includes('tst/')
+    const fileNameMatchTestPatterns = isTestFileByName(filePath, languageConfig.languageId)
 
-    // TODO (Metrics): Add total number of calls to isTestFile
-    if (isTestFileByName(editor.document.uri.fsPath, editor.document.languageId)) {
+    if (pathContainsTest || fileNameMatchTestPatterns) {
         return true
     }
 
-    // TODO (Metrics): Add metrics for isTestFileByName Failure
-    // (to help us determine if people follow naming conventions)
-    if (await dependencyGraph.isTestFile(editor.document.getText())) {
-        return true
-    }
-
-    return false
+    // This run slowly thus lazily execute
+    const fileHasTestDependency =
+        languageConfig.dependencyGraph && languageConfig.fileContent
+            ? await languageConfig.dependencyGraph.isTestFile(languageConfig.fileContent)
+            : false
+    return fileHasTestDependency
 }
```

Comment:
make dependencyGraph an optional graph since 
1. it's only used for 1 case which is unit test generator but not crossfile.
2. only a few simple APIs of it are used, and these APIs can be replaced with simple VSCode APIs.

Reply:
instead of making fake and hollow Typescript dependencyGraph object, making this parameter an optional is easier to maintain.